src/Components/Blazor/Build/src/Tasks/GenerateTypeGranularityLinkingConfig.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.AspNetCore.Blazor.Build.Tasks
+{
+    public class GenerateTypeGranularityLinkingConfig : Task
+    {
+        [Required]
+        public ITaskItem[] Assemblies { get; set; }
+
+        [Required]
+        public string OutputPath { get; set; }
+
+        public override bool Execute()
+        {
+            using (var fileStream = File.Open(OutputPath, FileMode.Create))
+            using (var xmlWriter = XmlWriter.Create(fileStream, new XmlWriterSettings { Indent = true }))
```

Comment:
Could this be changed to use `XElement` to be consistent with https://github.com/dotnet/aspnetcore/blob/master/src/Components/Blazor/Build/src/Tasks/BlazorCreateRootDescriptorFile.cs?

Reply:
Sure, done